src/Storage/ABS/SynapseImportAdapter.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\Db\ImportExport\Storage\ABS;
+
+use Doctrine\DBAL\Connection;
+use Keboola\Db\ImportExport\Backend\BackendImportAdapterInterface;
+use Keboola\Db\ImportExport\ImportOptions;
+use Keboola\Db\ImportExport\Storage\DestinationInterface;
+use Keboola\Db\ImportExport\Storage\Synapse\Table;
+use Keboola\Db\ImportExport\Storage\SourceInterface;
+
+class SynapseImportAdapter implements BackendImportAdapterInterface
+{
+    /**
+     * @var SourceFile
+     */
+    private $source;
+
+    /**
+     * @param SourceFile $source
+     */
+    public function __construct(SourceInterface $source)
+    {
+        $this->source = $source;
+    }
+
+    /**
+     * phpcs:disable SlevomatCodingStandard.TypeHints.TypeHintDeclaration.MissingParameterTypeHint
+     * @param Table $destination
+     * @param Connection $connection
+     */
+    public function getCopyCommands(
+        DestinationInterface $destination,
+        ImportOptions $importOptions,
+        string $stagingTableName,
+        $connection = null
+    ): array {
+
+        if ($connection === null || !$connection instanceof Connection) {
+            throw new \Exception(sprintf('Connection must be instance of "%s"', Connection::class));
+        }
+
+        $platform = $connection->getDatabasePlatform();
+        $sasToken = $this->source->getSasToken();
+        $destinationSchema = $platform->quoteSingleIdentifier($destination->getSchema());
+        $destinationTable = $platform->quoteSingleIdentifier($stagingTableName);
+
+        $fieldDelimiter = $connection->quote($this->source->getCsvOptions()->getDelimiter());
+        $firstRow = '';
+        if ($importOptions->getNumberOfIgnoredLines() !== 0) {
+            $firstRow = sprintf(',FIRSTROW=%s', $importOptions->getNumberOfIgnoredLines() + 1);
+        }
+        $enclosure = $connection->quote($this->source->getCsvOptions()->getEnclosure());
+
+        $entries = $this->source->getManifestEntries(SourceFile::PROTOCOL_HTTPS);
+
+        if (count($entries) === 0) {
+            return [];
+        }
+
+        foreach ($entries as &$entry) {
+            $entry = $connection->quote($entry);
+        }
+        unset($entry);
+
+        $entries = implode(', ', $entries);
+
+        return [
+            <<< EOT
+COPY INTO $destinationSchema.$destinationTable
+FROM $entries
```

Comment:
Je tam nějakej limit na počet entries?

Reply:
Nenašel sem takový limit.

Reply:
Asi to bude teda omezené velikostí query, což jestli to správně chápu bude 64kb https://docs.microsoft.com/en-us/sql/sql-server/maximum-capacity-specifications-for-sql-server?view=sql-server-ver15&viewFallbackFrom=azure-sqldw-latest
```

Length of a string containing SQL statements (batch size) |   | 65,536 (Network packet size)
-- | -- | --
Length of a string containing SQL statements (batch size) |   | 65,536 (Network packet size)
-- | -- | --
```
Takže snad ok pro začátek.
A koukám že to vlastně i testujeme na souboru s 1500 slicy (i když na prod ty cesty budou asi delší). V testu to projde ok, v query history ji vídím oříznutou asi na 4k znaků:
![image](https://user-images.githubusercontent.com/903531/77305955-34655700-6cf7-11ea-90fc-35493f1115e2.png)

Reply:
Jo právě prošel správně ten 1500 souborový import tak sem to moc neřešil, limit na počet souborů sem nenašel. Ikdyž je to preview, takže je možné že sa něco časem objeví.

Reply:
jo jasně, už jsem zapomněl že tam takový test je.